shared/src/price_estimate.rs

```diff
@@ -170,13 +183,24 @@ impl UniswapPriceEstimator {
         sell_token: H160,
         buy_token: H160,
         sell_amount: U256,
+        gas_price: f64,
     ) -> Result<(Vec<H160>, U256)> {
+        // Estimate with amount 0 to get a spot price (avoid potential endless recursion)
+        let buy_token_price_in_native_token = self
+            .estimate_price(self.native_token, buy_token, U256::zero(), OrderKind::Sell)
+            .await?;
         self.best_execution(
             sell_token,
             buy_token,
             sell_amount,
             |amount, path, pools| {
-                estimate_buy_amount(amount, path, pools).map(|estimate| estimate.value)
+                estimate_buy_amount(amount, path, pools).map(|estimate| {
+                    let proceeds_in_native_token =
+                        estimate.value.to_big_rational() * buy_token_price_in_native_token.clone();
+                    let tx_cost_in_native_token = U256::from_f64_lossy(gas_price).to_big_rational()
```

Comment:
ubernit: Given that native assets aren't actually tokens, I might propose names more like `native_tx_cost`, `native_proceeds`

Reply:
I'd say ETH is still a token (not an ERC20 though), wouldn't you agree?